improver_tests/calibration/dz_rescaling/test_apply_dz_rescaling.py

```diff
@@ -0,0 +1,218 @@
+    )
+    fp_coord = AuxCoord(
+        np.array(forecast_period * 3600, dtype=TIME_COORDS["forecast_period"].dtype,),
+        "forecast_period",
+        units=TIME_COORDS["forecast_period"].units,
+    )
+    time_coord = AuxCoord(
+        np.array(
+            pd.Timestamp(validity_time).timestamp(), dtype=TIME_COORDS["time"].dtype,
+        ),
+        "time",
+        units=TIME_COORDS["time"].units,
+    )
+    frt_coord = AuxCoord(
+        np.array(
+            pd.Timestamp(forecast_reference_time).timestamp(),
+            dtype=TIME_COORDS["forecast_reference_time"].dtype,
+        ),
+        "forecast_reference_time",
+        units=TIME_COORDS["forecast_reference_time"].units,
+    )
+
+    cube = build_spotdata_cube(
+        data,
+        "wind_speed_at_10m",
+        "m s-1",
+        altitude,
+        latitude,
+        longitude,
+        wmo_id,
+        scalar_coords=[fp_coord, time_coord, frt_coord],
+        additional_dims=[perc_coord],
+    )
+    return cube
+
+
+def _create_scaling_factor_cube():
+    """Create a scaling factor cube.
+
+    Returns:
+        Scaling factor cube.
+    """
+    cubelist = iris.cube.CubeList()
+    for frt_hour, multiplier in zip([3, 9], [0.1, 0.15]):
+        for fp_index, forecast_period in enumerate([6, 12, 18, 24]):
+            data = np.array([1, 1.1, 1, 1.05]) + fp_index * multiplier
+            altitude = np.array([0, 100, 20, 50])
+            latitude = np.array([0, 1, 2, 3])
+            longitude = np.array([0, 1, 2, 3])
+            wmo_id = ["00001", "00002", "00003", "00004"]
+            fp_coord = AuxCoord(
+                np.array(
+                    forecast_period * 3600, dtype=TIME_COORDS["forecast_period"].dtype,
+                ),
+                "forecast_period",
+                units=TIME_COORDS["forecast_period"].units,
+            )
+            frth_coord = AuxCoord(
+                np.array(frt_hour * 3600, dtype=np.int32,),
+                long_name="forecast_reference_time_hour",
+                units="seconds",
+            )
+            cube = build_spotdata_cube(
+                data,
+                "scaled_vertical_displacement",
+                "1",
+                altitude,
+                latitude,
+                longitude,
+                wmo_id,
+                scalar_coords=[fp_coord, frth_coord],
+            )
+            cubelist.append(cube)
+    return cubelist.merge_cube()
+
+
+@pytest.mark.parametrize("wmo_id", [True, False])
+@pytest.mark.parametrize(
+    "forecast_period,expected_data",
+    [
+        (5, [[0.0, 16.5, 5, 10.5], [0, 22, 10, 15.75], [5, 27.5, 15, 21]]),
+        (7, [[0.0, 18, 5.5, 11.5], [0, 24, 11, 17.25], [5.5, 30, 16.5, 23]]),
+        (11, [[0.0, 18, 5.5, 11.5], [0, 24, 11, 17.25], [5.5, 30, 16.5, 23]]),
+        (12, [[0.0, 18, 5.5, 11.5], [0, 24, 11, 17.25], [5.5, 30, 16.5, 23]]),
+    ],
+)
+def test_apply_dz_rescaling(wmo_id, forecast_period, expected_data):
```

Comment:
I don't feel that I have confidence that the method makes the correct adjustment. It would be easier to follow if all the inputs were 1 except the one you wanted to test, so that these tests test one number at a time, rather than 12. I should be able to see a forecast value, a scaling factor value and a vertical displacement value going in and one modified forecast value coming out. The parametrization can then cycle through different values to get all the behaviours needed.

As discussed, I will attempt to do this as a PR to your PR.